src/client.rs

```diff
@@ -46,7 +46,7 @@ impl<Http: HttpClient> Client<Http> {
         Client {
             host: host.into(),
             api_key: api_key.map(|key| key.into()),
-            http_client: Box::new(IsahcClient),
+            http_client: IsahcClinet,
```

Comment:
This could be an elementary problem...
It seems to me that you are fulfilling.
```
mismatched types
expected type parameter `Http`
           found struct `request::IsahcClinet`
```